### PR TITLE
Catalina log-parsing

### PR DESCRIPTION
**Pull request by EgorAfutin**
Log-parsing,
Tests,
Splitter parser functionality enhanced.

### CODE REVIEW

File: src/applications/log_parsing/catalina/driver.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    timezones_property = configuration.property("timezone.priority", "dic")
+
+    event_creator = EventCreator(
+        Metadata([
+            ConfigurableTimestampField("timestamp", timezone_name, timezones_property, "@timestamp"),
+            StringField("level"),
+            StringField("event_type"),
+            StringField("message")
+        ]),
+        SplitterParser(" ", True, 3)
```
Comment (halfwhite) [inline src/applications/log_parsing/catalina/driver.py]: A little improvement of readability can be made if literals became constants which names describes a purpose of them.
I really  don't remember what means True and 3.

File: src/common/log_parsing/list_event_creator/parsers/splitter_parser.py [hunk 2]
```diff
@@ -6,20 +6,24 @@ class SplitterParser:
     Parses event using split function
     """
 
-    def __init__(self, delimiter, is_trim=False):
+    def __init__(self, delimiter, is_trim=False, max_split=None):
```
Comment (halfwhite) [inline src/common/log_parsing/list_event_creator/parsers/splitter_parser.py]: A purpose is not obvious from the parameter name. A documentation could help.

File: src/test/unit/common/log_parsing/list_event_creator/parsers/test_splitter_parser.py [hunk 3]
```diff
@@ -15,6 +15,10 @@ def test_simple_split_parse_trim(self):
         parser = SplitterParser("|", is_trim=True)
         self.assertEquals(["a", "b", "c"], parser.parse("a   |    b    |     c"))
 
+    def test_split_parse_with_max_split(self):
```
Comment (halfwhite) [inline src/test/unit/common/log_parsing/list_event_creator/parsers/test_splitter_parser.py]: I'm not sure what is max_split, but obviously one test for it is not enough. 1 is a kind of boundary values.

File: src/applications/log_parsing/catalina/driver.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    timezones_property = configuration.property("timezone.priority", "dic")
+
+    event_creator = EventCreator(
+        Metadata([
+            ConfigurableTimestampField("timestamp", timezone_name, timezones_property, "@timestamp"),
+            StringField("level"),
+            StringField("event_type"),
+            StringField("message")
+        ]),
+        SplitterParser(" ", is_trim=True, max_split=4)
```
Comment (vunder-kind) [inline src/applications/log_parsing/catalina/driver.py]: Try to use Regex Parser for more flexible parsing.

File: src/applications/log_parsing/catalina/driver.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    event_creator = EventCreator(
+        Metadata([
+            ConfigurableTimestampField("timestamp", timezone_name, timezones_property, "@timestamp"),
+            StringField("level"),
+            StringField("event_type"),
+            StringField("thread"),
+            StringField("message")
+        ]),
+        RegexpParser(r"^(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+\+\d{4})\s+(?P<level>.*?)\s+"
```
Comment (vunder-kind) [inline src/applications/log_parsing/catalina/driver.py]: Change + to any symbol

File: src/applications/log_parsing/catalina/driver.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                     r"(?P<event_type>.*?)\s+\[(?P<thread>.*)\]\s(?P<message>.*)")
```
Comment (vunder-kind) [inline src/applications/log_parsing/catalina/driver.py]: Make thread ungreedy

File: src/test/unit/applications/log_parsing/catalina/test_catalina_log_parsing.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    event_creators = create_event_creators(Configuration(dict={"timezone": {"name": "Europe/Amsterdam"}}))
+
+    def test_catalina(self):
+        self.assert_parsing(
+            {
+                "source": "/penthera/logs/mongodb/mongo.log",
+                "message": "2018-04-11T07:49:19.072+0000 I NETWORK  [conn1564572] end connection 172.16.145.9:54914 (1141 connections now open)"
+            },
+            {
+                "@timestamp": datetime(2018, 4, 11, 7, 49, 19, 72000).replace(tzinfo=timezones["Europe/Amsterdam"]),
```
Comment (vunder-kind) [inline src/test/unit/applications/log_parsing/catalina/test_catalina_log_parsing.py]: Wrong time after parsing. Use "idc" priority in driver

File: src/applications/log_parsing/catalina/driver.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+def create_event_creators(configuration):
+    """
+    Method creates configuration for Catalina
+    :param configuration
+    :return: Composite event creator for Catalina
+    """
+
+    timezone_name = configuration.property("timezone.name")
+    timezones_property = configuration.property("timezone.priority", "dic")
```
Comment (vunder-kind) [inline src/applications/log_parsing/catalina/driver.py]: Use "idc" priority

File: src/applications/log_parsing/catalina/driver.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+from common.log_parsing.dict_event_creator.event_creator import EventCreator
+from common.log_parsing.log_parsing_processor import LogParsingProcessor
+from common.log_parsing.metadata import Metadata, StringField
+from common.log_parsing.timezone_metadata import ConfigurableTimestampField
+from util.utils import Utils
+
+
+def create_event_creators(configuration):
+    """
+    Method creates configuration for Catalina
```
Comment (halfwhite) [inline src/applications/log_parsing/catalina/driver.py]: Very confusing comment. Method creates configuration and it takes configuration as a parameter as well.
